bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/URIUtil.java

```diff
@@ -328,6 +330,33 @@ public static URI toURI(URL url) throws URISyntaxException {
 		}
 	}
 
+	/**
+	 * Returns the {@code Path} specified by the given {@code file} {@link URL}.
+	 * 
+	 * @throws IllegalArgumentException if the specified URL does not use the
+	 *                                  {@code file} protocol
+	 * 
+	 * @since 3.20
+	 */
+	public static Path toFilePath(URL url) {
+		if (!SCHEME_FILE.equals(url.getProtocol())) {
+			throw new IllegalArgumentException("Not a 'file' url: " + url); //$NON-NLS-1$
+		}
```

Comment:
Alternatively this method could return an empty optional if the URL is not a file-URL.
But since callers that are able to handle other protocols can simple check the precondition before calling this method, I think it's fine to keep it simple for those cases where a file-URL is used with certainty.

Reply:
Optional seem much nicer, calles can throw exception if the want anyways with `Optional.orElseThrow(...)`

Then we can easily handel other error conditions here as well for example if invalid characters are in the path.